openpecha/serializers/elan.py

```diff
@@ -0,0 +1,92 @@
+from pathlib import Path
+from openpecha.core.layer import LayerEnum
+from openpecha.serializers.serialize import Serialize
+
+
+def micro_to_milliseconds(microseconds):
+    return microseconds // 1000
+
+
+class ElanSerializer(Serialize):
+    """Elan serializer class to get .eaf file."""
+
+    def __init__(
+        self, opf_path, text_id=None, base_ids=None, layers=None, index_layer=None
+    ):
+        super().__init__(opf_path, text_id, base_ids, layers, index_layer)
+        self.annotation_sequence = {}
+        self.time_order = {}
+
+    def apply_annotation(self, base_id, annotation, uuid2localid):
+
+        ann_type = LayerEnum(annotation["type"])
+        if ann_type == LayerEnum.transcription_time_span:
+            if base_id not in self.annotation_sequence:
+                self.annotation_sequence[base_id] = 1
+            start_payload = """        <ANNOTATION>
+            <ALIGNABLE_ANNOTATION ANNOTATION_ID="a{seq}"
+                TIME_SLOT_REF1="ts{slot1}" TIME_SLOT_REF2="ts{slot2}">
+                <ANNOTATION_VALUE>""".format(
+                seq=self.annotation_sequence[base_id],
+                slot1=self.annotation_sequence[base_id] * 2 - 1,
+                slot2=self.annotation_sequence[base_id] * 2,
+            )
+            end_payload = """</ANNOTATION_VALUE>
+            </ALIGNABLE_ANNOTATION>
+        </ANNOTATION>"""
+            self.add_chars(base_id, annotation["span"]["start"], True, start_payload)
+            self.add_chars(base_id, annotation["span"]["end"], True, end_payload)
+            if base_id not in self.time_order:
+                self.time_order[
+                    base_id
+                ] = f"""        <TIME_SLOT TIME_SLOT_ID="ts{self.annotation_sequence[base_id] * 2 - 1}" TIME_VALUE="{micro_to_milliseconds(annotation["time_span"]["start"])}"/>
+        <TIME_SLOT TIME_SLOT_ID="ts{self.annotation_sequence[base_id] * 2}" TIME_VALUE="{micro_to_milliseconds(annotation["time_span"]["end"])}"/>"""
+            else:
+                self.time_order[
+                    base_id
+                ] += f"""
+        <TIME_SLOT TIME_SLOT_ID="ts{self.annotation_sequence[base_id] * 2 - 1}" TIME_VALUE="{micro_to_milliseconds(annotation["time_span"]["start"])}"/>
+        <TIME_SLOT TIME_SLOT_ID="ts{self.annotation_sequence[base_id] * 2}" TIME_VALUE="{micro_to_milliseconds(annotation["time_span"]["end"])}"/>"""
+            self.annotation_sequence[base_id] += 1
+
+    def get_elan(self, base_id, result):
+        return f"""<?xml version="1.0" encoding="UTF-8"?>
+<ANNOTATION_DOCUMENT AUTHOR="" DATE="2023-03-15T20:50:48+05:30"
+    FORMAT="3.0" VERSION="3.0"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.mpi.nl/tools/elan/EAFv3.0.xsd">
+    <HEADER MEDIA_FILE="" TIME_UNITS="milliseconds">
+        <MEDIA_DESCRIPTOR
+            MEDIA_URL="file:///Users/spsither/Documents/OpenPecha/OTR002-Output/OTR002-01-གངས་རིན་པོ་ཆེ།/OTR001.wav"
```

Comment:
How do I get the media_url from the TranscriptionTimeSpan layer and insert it here?

Reply:
I incorporated @10zinten 's suggestion  here